src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -315,6 +315,10 @@ public override ImmutableArray<CSharpAttributeData> GetAttributes()
 
         internal override bool IsMetadataOptional => _baseParameterForAttributes?.IsMetadataOptional == true;
 
+        internal override bool IsMetadataIn => RefKind == RefKind.In || _baseParameterForAttributes?.GetDecodedWellKnownAttributeData()?.HasInAttribute == true;
+
+        internal override bool IsMetadataOut => RefKind == RefKind.Out || _baseParameterForAttributes?.GetDecodedWellKnownAttributeData()?.HasOutAttribute == true;
```

Comment:
>_baseParameterForAttributes?.GetDecodedWellKnownAttributeData()?.HasOutAttribute == true [](http://example.com/codeflow?start=74&length=88)

I'd expect this change to be backed by a test (suggested in the comment that motivated the change).